### PR TITLE
peers: fix subnets index removal logic

### PR DESCRIPTION
**Pull request by nkryuchkov**
The logic is incorrect for `i != 0`, the peer is not removed.

This change is copied from https://github.com/ssvlabs/ssv/pull/1813 as it seems more important

### CODE REVIEW

File: network/peers/subnets_test.go [hunk 1]
```diff
@@ -120,3 +122,115 @@ func TestSubnetScore(t *testing.T) {
 		})
 	}
 }
+
+func TestUpdatePeerSubnets_Removal(t *testing.T) {
```
Comment (nkryuchkov) [inline network/peers/subnets_test.go:126]: This is likely not the final version of the tests, I'll be working on them on https://github.com/ssvlabs/ssv/pull/1813. I just wanted this PR to have some PoC of the fix

**Review by iurii-ssv — APPROVED** [review]
LGTM as long as tests cover edge cases (the code is quite involved and hence deserves thorough testing I think)

**Review by moshe-blox — APPROVED** [review]
good catch 💪
